### ISSUE TITLE
qt.accessibility.atspi: Error in contacting registry: "org.freedesktop.DBus.Error.NameHasNoOwner" "Could not activate remote peer 'org.a11y.atspi.Registry': unit failed"

### ISSUE THREAD

**Issue by jean-christophe-manciot** [bit-team/backintime#2025]
### Describe the problem, feature or ask a question:

This issue happens for all snapshots:
```
...
DEBUG: [qt/icon.py:20 <module>] Checking if the current theme contains the BiT icon...
DEBUG: [qt/icon.py:36 <module>] Probing theme: ubuntu-mono-dark (activated as ubuntu-mono-dark)
DEBUG: [qt/icon.py:32 <module>] Found an installed theme: ubuntu-mono-dark
qt.accessibility.atspi: Error in contacting registry: "org.freedesktop.DBus.Error.NameHasNoOwner" "Could not activate remote peer 'org.a11y.atspi.Registry': unit failed"
DEBUG: [common/snapshots.py:883 Snapshots.backup] Command "rsync --recursiv" returns 0
INFO: [common/snapshots.py:1079 Snapshots.backupConfig] Save config file
INFO: [common/snapshots.py:1160 Snapshots.backupPermissions] Save permissions
...
```

It does not prevent the snapshots from completing successfully though.

```
# backintime --diagnostics
{
    "backintime": {
        "name": "Back In Time",
        "version": "1.5.3",
        "running-as-root": false,
        "latest-config-version": 6,
        "local-config-file": "/root/.config/backintime/config",
        "local-config-file-found": true,
        "global-config-file": "/etc/backintime/config",
        "global-config-file-found": false,
        "started-from": "/usr/share/backintime/common",
        "user-callback": "/root/.config/backintime/user-callback",
        "keyring-supported": false
    },
    "host-setup": {
        "OS": {
            "/etc/os-release": "Ubuntu 24.10",
            "/etc/lsb-release": "DISTRIB_ID=Ubuntu\nDISTRIB_RELEASE=24.10\nDISTRIB_CODENAME=oracular\nDISTRIB_DESCRIPTION=\"Ubuntu 24.10\"\n",
            "/etc/debian_version": "trixie/sid\n"
        },
        "platform": "Linux-6.12.0-12-generic-x86_64-with-glibc2.40",
        "system": "Linux #12-Ubuntu SMP PREEMPT_DYNAMIC Wed Jan 22 16:36:37 UTC 2025",
        "display-system": "($XDG_SESSION_TYPE not set)",
        "locale": "en_US, UTF-8",
        "PATH": "/usr/lib/go/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/snap/bin",
        "RSYNC_OLD_ARGS": "(not set)",
        "RSYNC_PROTECT_ARGS": "(not set)"
    },
    "python-setup": {
        "python": "3.12.7 main Jan 17 2025 16:55:27 CPython GCC 14.2.0",
        "python-executable": "/usr/bin/python3",
        "python-executable-symlink": true,
        "python-executable-resolved": "/usr/bin/python3.12",
        "sys.path": [
            "/usr/share/backintime/qt/plugins",
            "/usr/share/backintime/common/plugins",
            "/usr/share/backintime/plugins",
            "/usr/share/backintime/common",
            "/usr/lib/python312.zip",
            "/usr/lib/python3.12",
            "/usr/lib/python3.12/lib-dynload",
            "/usr/local/lib/python3.12/dist-packages",
            "/usr/lib/python3/dist-packages",
            "/usr/lib/python3.12/dist-packages"
        ],
        "qt": {
            "Version": "PyQt 6.7.1 / Qt 6.6.2",
            "Theme": "hicolor",
            "Theme Search Paths": [
                "/root/.local/share/icons",
                "/usr/local/share/icons",
                "/usr/share/icons",
                ":/icons"
            ],
            "Fallback Theme": "hicolor",
            "Fallback Search Paths": [
                "/usr/share/pixmaps"
            ]
        }
    },
    "external-programs": {
        "rsync": {
            "version": "3.3.0",
            "protocol": "31.0",
            "capabilities": "file_bits: 64; inum_bits: 64; timestamp_bits: 64; long_int_bits: 64; socketpairs: True; symlinks: True; symtimes: True; hardlinks: True; hardlink_specials: True; hardlink_symlinks: True; IPv6: False; atimes: True; batchfiles: True; inplace: True; append: True; ACLs: True; xattrs: True; secluded_args: optional; iconv: True; prealloc: True; stop_at: True; crtimes: False",
            "optimizations": "SIMD_roll: True; asm_roll: False; openssl_crypto: True; asm_MD5: False",
            "checksum_list": "xxh128, xxh3, xxh64, md5, md4, sha1, none",
            "compress_list": "zstd, lz4, zlibx, zlib, none",
            "daemon_auth_list": "sha512, sha256, sha1, md5, md4"
        },
        "ssh": "OpenSSH_9.9p1 Ubuntu-3ubuntu2, OpenSSL 3.4.0 22 Oct 2024",
        "sshfs": "3.7.3",
        "encfs": "(no encfs)",
        "shell": "/bin/bash",
        "shell-version": "GNU bash, version 5.2.32(1)-release (x86_64-pc-linux-gnu)"
    }
}


```

**Comment by buhtz** [comment]
Hello Jean,

Thank you for taking the time to report the issue and providing the details. I appreciate your feedback.

I am assuming this is not related to BIT.

How do you start BIT in this cases where this error appears?

If you have any more details to share, feel free to reach out.

Best regards,
Christian

**Comment by jean-christophe-manciot** [comment]
> How do you start BIT in this cases where this error appears?

```
/usr/bin/nice -n 19 /usr/bin/ionice -c2 -n7 /usr/bin/backintime --debug --profile "$profile_name" backup
```

**Comment by buhtz** [comment]
Beside that error output itself I don't see that the backup job itself is somehow interrupted.

> > How do you start BIT in this cases where this error appears?
> 
> ```
> /usr/bin/nice -n 19 /usr/bin/ionice -c2 -n7 /usr/bin/backintime --debug --profile "$profile_name" backup
> ```

Looks like a crontab line. Can you reproduce the error output with exactly that line in a shell?

Can you also reproduce the error output when you start `backintime-qt` in terminal?

I just can tell that `backintime` has no Qt related dependency but try to use Qt in [`qt_probing.py`](https://github.com/bit-team/backintime/blob/dev/common/qt_probing.py) just because of debug reasons. So I am assuming that Qt is somehow activated around this code.

But I still assume this is not BIT related but Qt. I recommend to ask the Ubuntu and Qt community.

If this error is restricted to the cron environment only, than I assume that the accessibility components are somehow not active or available to Qt when running via cron.

**Comment by jean-christophe-manciot** [comment]
> Looks like a crontab line. Can you reproduce the error output with exactly that line in a shell?
> 
> Can you also reproduce the error output when you start `backintime-qt` in terminal?

I wasn't clear enough. This error pops up only when I run the above command from CLI (bash); when it is automatically activated from crontab, there is no such error.

**Comment by buhtz** [comment]
Interesting. I really have no idea. Let me know if get some more info from somewhere else about it.

**Comment by buhtz** [comment]
I do ~~prepare~~ have a [release candidate](https://github.com/bit-team/backintime/releases/tag/v1.5.4-rc1) targeting ~~1.6.0~~1.5.4. Let's see if someone else also got this error.

**Comment by buhtz** [comment]
Closing this ticket based on the comment above. Feel free to reopen
if the problem still exists. Thank you for your efforts. If you have
any further questions, ideas or encounter any other issues, please
don't hesitate to let us know.

Best regards,
Christian
